### PR TITLE
Rename Territory -> Region for Displayname component

### PR DESCRIPTION
**Pull request by snktd**
This is based on the discussion here https://github.com/unicode-org/icu4x/pull/2816#discussion_r1017193736.

### CODE REVIEW

File: provider/datagen/src/transform/cldr/cldr_serde/region_displaynames.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// This file is part of ICU4X. For terms of use, please see the file
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/cldr_serde/region_displaynames.rs:1]: Nit: It looks like you added `region_displaynames.rs` but didn't delete `territory_displaynames.rs`

File: provider/datagen/src/transform/cldr/displaynames/region.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
+// This file is part of ICU4X. For terms of use, please see the file
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/displaynames/region.rs:1]: I also suspect you added region.rs without deleting territory.rs

**Review by sffc — CHANGES_REQUESTED** [review]

File: provider/datagen/src/transform/cldr/displaynames/region.rs (hunk 2 above)
Comment (snktd) [reply provider/datagen/src/transform/cldr/displaynames/region.rs]: Yes. Forgot to commit the change. Removed now.

File: provider/datagen/src/transform/cldr/cldr_serde/region_displaynames.rs (hunk 1 above)
Comment (snktd) [reply provider/datagen/src/transform/cldr/cldr_serde/region_displaynames.rs]: Done.

**Review by sffc — DISMISSED** [review]

**Review by robertbastian — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**